sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -196,12 +196,10 @@ case class FileSourceScanExec(
     optionalNumCoalescedBuckets: Option[Int],
     dataFilters: Seq[Expression],
     tableIdentifier: Option[TableIdentifier],
-    disableBucketedScan: Boolean = false)
+    disableBucketedScan: Boolean = false,
+    metadataColumns: Seq[AttributeReference])
```

Comment:
Now we can avoid adding this new parameter. When convert the fields of struct-type metadata col, we can use `FileSourceMetadataAttribute`

Reply:
oops, forgot to revert it back...